src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/ReadsPipelineSpark.java

```diff
@@ -111,7 +114,18 @@ protected void runTool(final JavaSparkContext ctx) {
         //NOTE: this doesn't honor enabled/disabled commandline filters
         final ReadFilter bqsrReadFilter = ReadFilter.fromList(BaseRecalibrator.getBQSRSpecificReadFilterList(), getHeaderForReads());
 
-        final JavaRDD<GATKRead> markedFilteredReadsForBQSR = markedReads.filter(read -> bqsrReadFilter.test(read));
+        JavaRDD<GATKRead> markedFilteredReadsForBQSR = markedReads.filter(read -> bqsrReadFilter.test(read));
+
+        if (joinStrategy.equals(JoinStrategy.OVERLAPS_PARTITIONER)) {
+            // the overlaps partitioner requires that reads are coordinate-sorted
+            final SAMFileHeader readsHeader = getHeaderForReads().clone();
+            readsHeader.setSortOrder(SAMFileHeader.SortOrder.coordinate);
+            ReadCoordinateComparator comparator = new ReadCoordinateComparator(readsHeader);
+            markedFilteredReadsForBQSR = markedFilteredReadsForBQSR
+                    .mapToPair(read -> new Tuple2<>(read, null))
```

Comment:
Also, can you extract a utility method in `SparkUtils` that sorts an RDD of reads by coordinate? This seems like something that would be generally useful.